integration/tests/cook/test_basic.py

```diff
@@ -213,19 +215,34 @@ def test_max_runtime_exceeded(self):
         job_timeout_interval_ms = job_timeout_interval_seconds * 1000
         max_runtime_ms = 5000
         assert max_runtime_ms < job_timeout_interval_ms
-        job_uuid, resp = util.submit_job(self.cook_url, command='sleep %s' % job_timeout_interval_seconds,
-                                         executor=job_executor_type, max_runtime=max_runtime_ms)
-        self.assertEqual(201, resp.status_code, msg=resp.content)
-        util.wait_for_job(self.cook_url, job_uuid, 'completed', job_timeout_interval_ms)
-        job = util.load_job(self.cook_url, job_uuid)
-        self.assertEqual(1, len(job['instances']))
-        self.assertEqual('failed', job['instances'][0]['status'])
-        self.assertEqual(2003, job['instances'][0]['reason_code'])
-        self.assertEqual('Task max runtime exceeded', job['instances'][0]['reason_string'])
-        if job_executor_type == 'cook':
-            job = util.wait_for_exit_code(self.cook_url, job_uuid)
-            self.assertEqual(-15, job['instances'][0]['exit_code'])
-            self.assertTrue(bool(job['instances'][0]['sandbox_directory']))
+        try:
+            job_uuid, resp = util.submit_job(self.cook_url,
+                                             command=f'sleep {job_timeout_interval_seconds}',
+                                             executor=job_executor_type, max_runtime=max_runtime_ms)
+            self.assertEqual(201, resp.status_code, msg=resp.content)
+            util.wait_for_job(self.cook_url, job_uuid, 'running')
+            util.wait_for_job(self.cook_url, job_uuid, 'completed', job_timeout_interval_ms)
+            job = util.wait_for_end_time(self.cook_url, job_uuid)
```

Comment:
Do we need to wait for running, then wait for completed, then wait for end time? Seems like we could just have the wait for end time, no?

Reply:
The wait-for-completed has a timeout. My worry is that if there's a delay scheduling the job, then the timeout could fail and give us a false-negative. There's also a weird delay waiting for the `end_time` to show up... Not sure what causes that. This is more verbose, but it seemed more robust. If we just had a single timeout, it wouldn't be clear if it failed because 1) the job was delayed being scheduled, 2) the job wasn't cancelled when we expected, or 3) the `end_time` didn't show up in a timely manner.

Reply:
What do you think about adding some of the explanation above into comments for the wait for running, wait for completed, wait for end time lines?